src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/ReferenceMessageRecep.java

```diff
@@ -0,0 +1,57 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
+
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+public class ReferenceMessageRecep extends Reference {
+
+    public static final String QUALIFIER = "MIS";
+
+    private @NonNull Long messageSequenceNumber;
+    private @NonNull RecepCode recepCode;
+
+    private String buildReferenceString() {
+        return messageSequenceNumber + " " + recepCode.name();
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+        if (StringUtils.isEmpty(getReference())) {
```

Comment:
This should validate each of the segment's data items separately. Is "NaN CP" is valid reference?